File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

convert(N) -> convert(N, [{N rem 3, "Pling"}, {N rem 5, "Plang"}, {N rem 7, "Plong"}], []).

convert(N, [{0, Drop}|T], Acc) -> convert(N, T, [Drop|Acc]);
convert(N, [_|T], Acc) -> convert(N, T, Acc);
convert(N, [], []) -> convert(N, [], [N]);
convert(_, [], Acc) -> lists:concat(lists:reverse(Acc)).

File: julia/raindrops/raindrops.jl
function raindrops(number)
    res = []
    number % 3 == 0 && push!(res, "Pling")
    number % 5 == 0 && push!(res, "Plang")
    number % 7 == 0 && push!(res, "Plong")
    res == [] && push!(res, string(number))
    join(res, "")
end
